pkg/kapp/clusterapply/delete_change.go

```diff
@@ -32,10 +33,40 @@ type DeleteChange struct {
 	identifiedResources ctlres.IdentifiedResources
 }
 
+type uniqueResourceRef struct {
+	schema.GroupKind
+	Name string
+}
+
+// if any new reource will encountered which kapp can not delete and required to orphaned then resource info will be added here.
+var inoperableResourceList = []uniqueResourceRef{
+	{
+		GroupKind: schema.GroupKind{Group: "", Kind: "namespace"},
+		Name:      "default",
+	},
+	{
+		GroupKind: schema.GroupKind{Group: "", Kind: "namespace"},
+		Name:      "kube-node-lease",
+	},
+	{
+		GroupKind: schema.GroupKind{Group: "", Kind: "namespace"},
+		Name:      "kube-public",
+	},
+	{
+		GroupKind: schema.GroupKind{Group: "", Kind: "namespace"},
+		Name:      "kube-system",
+	},
+}
+
 func (c DeleteChange) ApplyStrategy() (ApplyStrategy, error) {
 	res := c.change.ExistingResource()
+
 	strategy := res.Annotations()[deleteStrategyAnnKey]
 
```

Comment:
nit: do not think we need this new line

Reply:
thanks for pointing this out, removed it.